src/Daemon/Daemon.cpp

```diff
@@ -460,6 +466,51 @@ namespace usbguard
   {
   }
 
+  pid_t Daemon::pid_fd = -1;
+
+  void Daemon::daemonize(const std::string &pid_file)
+  {
+      USBGUARD_LOG(Trace) << "Starting daemonization";
+
+      pid_t pid = 0;
+
+      USBGUARD_SYSCALL_THROW("Daemonize", (pid = fork()) < 0);
+      if (pid > 0) {
+        usleep(2500); /* We need to die AFTER pidfile is written to... */
```

Comment:
Because we need the first parent to exit after second child writes PID to the file which is complicated to programatically ensure (if you have some idea please offer), unfortunately systemd reads pid file immediately after parent exits so some kind of delay is necessary. The amount can be adjusted (two forks, setsid and /var file write should be pretty fast but some compromise between reliability and not slowing start much is definitely needed).